### ISSUE TITLE
The database gets locked using version 2.2.0+2

### ISSUE THREAD

**Issue by UCetinkaya94** [tekartik/sqflite#893]
Hi, I just upgraded to `2.2.0+2` and started seeing the `database has been locked` warning. It seems that running multiple transactions causes this issue. 

The following code will lock the database using `sqflite: ^2.2.0+2` but it will run just fine using `2.1.0+1`

```dart
import 'package:flutter/material.dart';
import 'package:path/path.dart';
import 'package:sqflite/sqflite.dart';

void main() async {
  WidgetsFlutterBinding.ensureInitialized();

  final databasePath = await getDatabasesPath();
  String path = join(databasePath, 'my_db.db');

  final database = await openDatabase(
    path,
    version: 1,
    onCreate: (db, _) async {
      await db.execute(
        'CREATE TABLE Test (id INTEGER PRIMARY KEY, name TEXT)',
      );
    },
  );

  runApp(
    MaterialApp(
      home: HomeScreen(
        database: database,
      ),
    ),
  );
}

class HomeScreen extends StatelessWidget {
  const HomeScreen({
    super.key,
    required this.database,
  });

  final Database database;

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        backgroundColor: Colors.red,
      ),
      floatingActionButton: FloatingActionButton(
        onPressed: () async {
          for (int i = 0; i < 200; i++) {
            database.transaction(
              (txn) async {
                await txn.rawInsert(
                  'INSERT INTO Test (name) VALUES (?)',
                  ['test'],
                );

                print('inserted');
              },
            );
          }
        },
      ),
    );
  }
}
 ```

**Comment by alextekartik** [comment]
Thank you for the report and a good way to reproduce the issue. I was indeed able to reproduce the issue on Android which is pretty bad (and bad that I did not catch that with integration testing that I started to put in place). iOS/MacOS and ffi looks ok though. I'm on it right now.

**Comment by alextekartik** [comment]
Fixed in 2.2.0+3 just published. In fact I wonder why it worked before even for a small number of transactions (in my testing it regularly failed at the 126th transaction...). Thanks again for taking the time to write a small example that helped me investigate (and i derived an integration test from it) and sorry for the mistake.

**Comment by UCetinkaya94** [comment]
You're welcome and thanks for the fix!
